app/controllers/species_controller.rb

```diff
@@ -0,0 +1,56 @@
+class SpeciesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_species, except: [:index, :new, :create, :upload]
```

Comment:
I think methods that actually need `@species` variable are less than the ones who don't need it. If that's so change this before_action to:

```ruby
before_action :set_species, only: [:edit, :show]
```